### ISSUE TITLE
[UWP] Unable to cast object of type 'Microsoft.Graphics.Canvas.UI.Xaml.CanvasImageSource' to type 'Windows.UI.Xaml.Media.Imaging.BitmapSource'.

### ISSUE THREAD

**Issue by SuNNjek** [xamarin/Xamarin.Forms#5666]
### Description
Using ImageButton in combination with FontImageSource crashes the application under UWP.
The exception is the following:

```
System.InvalidCastException: Unable to cast object of type 'Microsoft.Graphics.Canvas.UI.Xaml.CanvasImageSource' to type 'Windows.UI.Xaml.Media.Imaging.BitmapSource'.
   at Xamarin.Forms.Platform.UWP.TaskExtensions.<>c.<WatchForError>b__2_1(Object e)
   at System.Threading.WinRTSynchronizationContextBase.Invoker.InvokeCore()
```

### Steps to Reproduce

1. Start the attached application
2. The app crashes instantly

### Expected Behavior
The app should show an ImageButton with the specified Material-Design icon as the source.

### Actual Behavior
The app crashes.

### Basic Information

- Version with issue: 3.6.0.264807
- Last known good version: -
- IDE: Visual Studio 15.9.9
- Platform Target Frameworks: 
  - UWP:  16299
- Nuget Packages: Only the packages added by the project template. No new packages were added

### Reproduction Link

[FontImageCrash.zip](https://github.com/xamarin/Xamarin.Forms/files/2996084/FontImageCrash.zip)

**Comment by SuNNjek** [comment]
In case it helps:
When fiddling around with the FontImageSource I got the same exception with a more verbose stack trace:
```
   at Xamarin.Forms.Platform.UWP.ImageButtonRenderer.GetDesiredSize(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.Platform.UWP.Platform.GetNativeSize(VisualElement element, Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.Platform.UWP.WindowsBasePlatformServices.GetNativeSize(VisualElement view, Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.VisualElement.OnSizeRequest(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.ImageButton.OnMeasure(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.VisualElement.GetSizeRequest(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.VisualElement.Measure(Double widthConstraint, Double heightConstraint, MeasureFlags flags)
   at Xamarin.Forms.Grid.CalculateAutoCells(Double width, Double height)
   at Xamarin.Forms.Grid.MeasureGrid(Double width, Double height, Boolean requestSize)
   at Xamarin.Forms.Grid.OnSizeRequest(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.VisualElement.OnMeasure(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.VisualElement.GetSizeRequest(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.Layout.GetSizeRequest(Double widthConstraint, Double heightConstraint)
   at Xamarin.Forms.VisualElement.Measure(Double widthConstraint, Double heightConstraint, MeasureFlags flags)
   at Xamarin.Forms.Platform.UWP.ViewToRendererConverter.WrapperControl.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.FrameworkElement.MeasureOverride(Size availableSize)
   at Xamarin.Forms.Platform.UWP.CellControl.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
   at Windows.UI.Xaml.UIElement.Measure(Size availableSize)
   at Xamarin.Forms.Platform.UWP.VisualElementRenderer`2.MeasureOverride(Size availableSize)
```

**Comment by paymicro** [comment]
@SuNNjek thanks. The issue is clear. PR https://github.com/xamarin/Xamarin.Forms/pull/5678 will fix it.

**Comment by samhouts** [comment]
closed by #5678
